dask_jobqueue/core.py

```diff
@@ -342,7 +342,7 @@ def stop_workers(self, workers):
         logger.debug("Stopping workers: %s" % workers)
         if not workers:
             return
-        jobs = self._stop_pending_jobs()  # stop pending jobs too
+        jobs = self._del_pending_jobs()  # stop pending jobs too
```

Comment:
I'm actually a bit confused by this line of code.  We always stop/delete pending jobs whenever we call `stop_workers`?  What if someone wants to stop a particular job even while other jobs are coming online?  

I know that this isn't related to this PR.  My apologies for the off-topic conversation.

Reply:
I was a really surprised at first too. But actually, if I understand correctly, `scale_down` will only be called if we want less workers that we currently have, so less worker that what is already online. This means that all pending jobs are useless and can be safely cancelled. 

On the other hand, we could call scale_up, but find that we already have enough pending jobs to fulfil the requested number. In this case we do nothing. Maybe we should even stop some of pending jobs if the new requested scale is less than running + pending.

I don't know if this is the correct approach regarding dask behaviour upstream. Maybe we should always consider the potential amount of worker as online ones... That doesn't sound really good though.

Any thought?

Reply:
And for argumentation about the name change, it is because _stop_pending_jobs is not actually stopping anything, so I wanted to make that clearer. Any suggestion on the name of the method welcome.